src/raftstore/store/store.rs

```diff
@@ -1691,17 +1691,14 @@ impl<T: Transport, C: PdClient> Store<T, C> {
     }
 
     fn handle_snap_mgr_gc(&mut self) -> Result<()> {
-        let mut snap_keys = try!(self.snap_mgr.list_snap());
+        let mut snap_keys = try!(self.snap_mgr.list_idle_snap());
```

Comment:
It needs this `delete_snapshot()` to be implemented like
https://github.com/pingcap/tikv/blob/hhkbp2/refactor-snapshot-manager/src/raftstore/store/snap.rs#L2065

to address 

https://github.com/pingcap/tikv/pull/1883#discussion_r120047665

So we couldn't remove the lock in `delete_snapshot()`, it's better to keep it this way.